src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -246,6 +237,79 @@ public static function register($mode = 0)
         }
     }
 
+    private static function isLaggingVendor(array $trace): bool
+    {
+        foreach ($trace as $line) {
+            if (isset($line['file'])) {
+                $file = $line['file'];
+                if ('-' === $file) {
+                    continue;
+                }
+                if (!self::inVendors($file)) {
+                    return false;
+                }
+                if (isset($erroringFile, $erroringPackage)) {
+                    if (self::getPackage($file) !== $erroringPackage) {
+                        return true;
+                    }
+                } else {
+                    $erroringFile = $file;
+                    $erroringPackage = self::getPackage($file);
+                }
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * inVendors() should always be called prior to calling this method.
+     */
+    private static function getPackage(string $path): string
+    {
+        $path = realpath($path) ?: $path;
+        foreach (self::getVendors() as $vendorRoot) {
+            if (0 === strpos($path, $vendorRoot)) {
+                $relativePath = substr($path, strlen($vendorRoot) + 1);
+                $vendor = strstr($relativePath, DIRECTORY_SEPARATOR, true);
+
+                return $vendor.'/'.strstr(substr($relativePath, strlen($vendor) + 1), DIRECTORY_SEPARATOR, true);
+            }
+        }
+    }
+
+    private static function getVendors(): ?array
+    {
+        /** @var string[] absolute paths to vendor directories */
+        static $vendors;
+
+        if (null === $vendors) {
+            foreach (get_declared_classes() as $class) {
```

Comment:
you need to initialize `$vendors` to an empty array before the loop. Otherwise, it is possible that `null` is returned instead of an empty array if no composer autoloader can be detected.